worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -306,10 +365,27 @@ private int processHandles(List<Future<Integer>> futureHandles) {
             }
         }
 
+        trackProgress(progressTracker);
+
         return errorCount;
     }
 
 
+    private void trackProgress(ProgressTracker progressTracker) {
+        var jobUuid = progressTracker.getJobUuid();
+        var processedCount = progressTracker.getProcessedCount();
+        var totalCount = progressTracker.getTotalCount();
+
+        final int percentageCompleted = (processedCount * 100) / totalCount;
+
+        if (processedCount - progressTracker.getLastUpdatedCount() >= reportProgressFrequency) {
+            progressTracker.setLastUpdatedCount(processedCount);
+            jobRepository.updatePercentageCompleted(jobUuid, percentageCompleted);
+            log.info("[{}/{}] records processed = [{}% completed]", processedCount, totalCount, percentageCompleted);
```

Comment:
We should update the job record if the _percentage_ has actually changed. Otherwise we may be issuing many `UPDATE` statements that actually don't change anything.

Should be simply a matter of adding `lastPercentageCompleted` to the tracker.

Reply:
done.
Added a `lastPercentageCompleted` & a separate `reportProgressLogFrequency` for logging the progress in the log.